codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/RustTypes.kt

```diff
@@ -375,6 +375,28 @@ sealed class Attribute {
                 writer.addDependency(it.dependency)
             }
         }
+
+        companion object {
+            /**
+             * Renders a
+             * [`#[deprecated]`](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute)
+             * attribute.
+             */
+            fun deprecated(note: String = "", since: String = ""): Custom {
+                val builder = StringBuilder()
+                builder.append("deprecated")
+                if (note.isBlank().and(since.isBlank())) {
+                    // No-op. Rustc would emit a default message.
+                } else if (note.isNotBlank().and(since.isBlank())) {
+                    builder.append("(note = \"${note}\")")
+                } else if (note.isBlank().and(since.isNotBlank())) {
+                    builder.append("(since = \"${since}\")")
+                } else {
+                    builder.append("(note = \"${note}\", since = \"${since}\")")
+                }
```

Comment:
Escaped quotes can be handled using the `dp()` function:

```kotlin
import software.amazon.smithy.rust.codegen.util.dq

} else if (note.isNotBlank().and(since.isBlank())) {
    builder.append("(note = "${note.qt()}")")
} else if (note.isBlank().and(since.isNotBlank())) {
    builder.append("(since = ${since.dq()})")
} else {
    builder.append("(note = ${note.dq()}, since = ${since.dq()})")
}
```